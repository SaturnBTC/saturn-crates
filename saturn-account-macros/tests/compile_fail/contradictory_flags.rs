use saturn_account_macros::Accounts;

#[derive(Accounts)]
struct ContradictoryFlags<'info> {
    #[account(zero_copy, init, payer = payer, program_id = arch_program::pubkey::Pubkey::default())]
    acc: arch_program::account::AccountInfo<'info>,
    #[account(signer)]
    payer: arch_program::account::AccountInfo<'info>,
}

fn main() {} 