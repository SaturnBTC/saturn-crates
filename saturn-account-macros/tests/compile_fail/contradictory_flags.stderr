error[E0599]: the function or associated item `new` exists for struct `AccountLoader<'_, AccountInfo<'info>>`, but its trait bounds were not satisfied
 --> tests/compile_fail/contradictory_flags.rs:3:10
  |
3 | #[derive(Accounts)]
  |          ^^^^^^^^ function or associated item cannot be called on `AccountLoader<'_, AccountInfo<'info>>` due to unsatisfied trait bounds
  |
 ::: $WORKSPACE/deps/program/src/account.rs
  |
  | pub struct AccountInfo<'a> {
  | -------------------------- doesn't satisfy `AccountInfo<'info>: bytemuck::pod::Pod` or `AccountInfo<'info>: bytemuck::zeroable::Zeroable`
  |
  = note: the following trait bounds were not satisfied:
          `AccountInfo<'info>: bytemuck::pod::Pod`
          `AccountInfo<'info>: bytemuck::zeroable::Zeroable`
  = note: this error originates in the derive macro `Accounts` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `AccountInfo<'info>: bytemuck::pod::Pod` is not satisfied
 --> tests/compile_fail/contradictory_flags.rs:3:10
  |
3 | #[derive(Accounts)]
  |          ^^^^^^^^ the trait `bytemuck::pod::Pod` is not implemented for `AccountInfo<'info>`
  |
  = help: the following other types implement trait `bytemuck::pod::Pod`:
            ()
            ManuallyDrop<T>
            PhantomData<T>
            PhantomPinned
            RuneAmount
            RuneId
            Wrapping<T>
            [T; N]
          and $N others
note: required by a bound in `AccountLoader`
 --> $WORKSPACE/saturn-account-parser/src/codec/zero_copy.rs
  |
  | pub struct AccountLoader<'a, T>
  |            ------------- required by a bound in this struct
  | where
  |     T: Pod + Zeroable + 'static,
  |        ^^^ required by this bound in `AccountLoader`
  = note: this error originates in the derive macro `Accounts` (in Nightly builds, run with -Z macro-backtrace for more info)
